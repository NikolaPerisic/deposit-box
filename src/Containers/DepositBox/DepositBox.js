import React from "react";
import "./DepositBox.scss";
import Display from "../../Components/Display/Display";
import Keypad from "../../Components/Keypad/Keypad";
import { connect } from "react-redux";
import * as actions from "../../actions/actions";
//
class DepositBox extends React.Component {
  //
  _submitTimer = null;
  _backlightTimer = null;
  //
  componentDidMount() {
    document.addEventListener("keydown", el => this.handleKeyPress(el.key));
    this.handleDisplayBacklight();
  }

  componentWillUnmount() {
    document.removeEventListener("keydown", el => this.handleKeyPress(el.key));
  }

  componentDidUpdate(prevState) {
    console.log(this.props.isBusy);
    if (this.props.isTouched) {
      this._submitTimer = setTimeout(() => {
        this.props.handleAutoSubmit(
          this.props.code,
          this.props.displayMsg,
          this.props.isLocked,
          this.props.isBusy
        );
      }, 2000);
    }
    clearTimeout(this._backlightTimer);
    this.handleDisplayBacklight();
  }
  // function for dimming the display after 5 secs of inactivity
  handleDisplayBacklight = () => {
    this._backlightTimer = setTimeout(() => {
      this.props.handleBacklight();
    }, 5000);
  };
  handleKeyPress = el => {
    if (!this.props.isTouched) {
      this.props.reset();
    }
    if (el === "l" || el === "L") {
      return this.props.handleLockFromUser(
        this.props.displayMsg,
        this.props.isLocked
      );
    }
    if ((!isNaN(el) && el !== " ") || el === "*") {
      clearTimeout(this._submitTimer);
      clearTimeout(this._backlightTimer);
      return this.props.handleUserInput(el);
    }
  };

  render() {
    return (
      <div className="panel">
        <Display
          isLocked={this.props.isLocked}
          message={this.props.displayMsg}
          backlight={this.props.backlightOn}
        />
        <Keypad clickInput={this.handleKeyPress} />
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    isLocked: state.depositBox.isLocked,
    displayMsg: state.depositBox.displayMsg,
    isTouched: state.depositBox.isTouched,
    isBusy: state.depositBox.isBusy,
    backlightOn: state.depositBox.backlightOn,
    code: state.depositBox.code
  };
};
const mapDispatchToProps = dispatch => {
  return {
    handleUserInput: value => dispatch(actions.handleUserInput(value)),
    handleLockFromUser: (userInput, status) =>
      dispatch(actions.handleLockFromUser(userInput, status)),
    reset: () => dispatch(actions.reset()),
    handleBacklight: () => dispatch(actions.handleBacklight()),
    handleAutoSubmit: (code, userInput, status, isBusy) =>
      dispatch(actions.handleAutoSubmit(code, userInput, status, isBusy))
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(DepositBox);
