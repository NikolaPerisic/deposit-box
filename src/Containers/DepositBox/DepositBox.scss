.panel {
  margin: 20px 0;
  position: relative;
  color: #fff;
  background-color: #7d7d7f;
  user-select: none;
  width: 400px;
  height: 600px;
  padding: 1.5rem 3rem 2.5rem 3rem;
  border: 2px solid black;
  border-radius: 5px;
  display: grid;
  grid-gap: 2rem;
  grid-template-columns: repeat(3, 1fr);
  grid-template-rows: repeat(6, 1fr);
  grid-template-areas:
    "a a a"
    "a a a"
    "b b b"
    "b b b"
    "b b b"
    "b b b";
}
@media (max-width: 680px) {
  .panel {
    margin: 0;
    width: 80vw;
  }
}
.panel__serial {
  font-size: 0.8rem;
  position: absolute;
  bottom: 10px;
  right: 15px;
  color: #434343;
}
