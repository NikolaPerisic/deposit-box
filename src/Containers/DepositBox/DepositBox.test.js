import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { DepositBox } from "./DepositBox";

//props
const props = {
  isLocked: false,
  displayMsg: "Ready",
  isTouched: false,
  isBusy: false,
  backlightOn: true,
  code: null,
  serial: null,
  service: false,
  lastKey: null
};

describe("Deposit Box", () => {
  const wrapper = shallow(<DepositBox {...props} />);
  //
  it("renders without crashing given req props", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
