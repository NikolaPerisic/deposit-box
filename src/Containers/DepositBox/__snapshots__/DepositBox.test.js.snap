// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deposit Box renders without crashing given req props 1`] = `
<div
  className="panel"
>
  <display
    backlight={true}
    isLocked={false}
    message="Ready"
  />
  <keypad
    activeItem={null}
    clickInput={[Function]}
  />
  <div
    className="panel__serial"
  >
    S/N: 
  </div>
</div>
`;
